### PR TITLE
Floating image issue 10

### PR DESCRIPTION
**Pull request by DanCollinz**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://floating-image-issue-10--takeda-ihs--hlxsites.hlx.page/drafts/floating-images
- After: https://floating-image-issue-10--takeda-ihs--hlxsites.hlx.page/drafts/floating-images

### CODE REVIEW

**Comment by bstopp** [comment]
Go ahead and add the necessary global styles to get the header, font, and page width management to this PR. That way the draft page for this auto-block looks like the live page.

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,7 @@ logs/*
 node_modules/*
 
 helix-importer-ui
-.DS_Store
\ No newline at end of file
+.DS_Store
+.idea/
+.vscode/
+styles/styles-old.css
```
Comment (lwintch) [inline .gitignore:10]: When ready, we should delete the `styles/styles-old.css` file then delete this line before merging.

File: scripts/scripts.js [hunk 2]
```diff
@@ -30,13 +30,48 @@ function buildHeroBlock(main) {
   }
 }
 
+/**
+* Builds floating image block and prepends to main in a new section.
+* @param {Element} main The container element
+*/
+function buildFloatingImages(main) {
+  main.querySelectorAll('.section-metadata').forEach((metadata) => {
+    let style;
```
Comment (lwintch) [inline scripts/scripts.js:39]: Looks like this could have a scoping issue.  I don't see this `style` being assigned a value, and the `other style` inside the `every` function call is in a closure.  This might always be uninitialized right now.

File: scripts/scripts.js [hunk 3]
```diff
@@ -30,13 +30,48 @@ function buildHeroBlock(main) {
...
+    [...metadata.querySelectorAll(':scope > div')].every((div) => {
+      const match = div.children[1]?.textContent.toLowerCase().trim().match(/(image-(left|right))/);
+      if (div.children[0]?.textContent.toLowerCase().trim() === 'style' && match) {
+        [, style] = match;
+        return false;
+      }
+      return true;
+    });
+    if (style) {
```
Comment (lwintch) [inline scripts/scripts.js:48]: Might always get skipped.  See above comment.

File: blocks/floating-images/floating-images.css [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+.block.floating-images > div {
+    display: flex;
+    flex-direction: column-reverse;
+    padding-top: 1em;
+    min-height: 350px;
+}
+
+.section.floating-images-container {
```
Comment (bstopp) [inline blocks/floating-images/floating-images.css:14]: This is outside the scope of the block, so shouldn't be here.

File: blocks/floating-images/floating-images.css [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+.block.floating-images .image img {
+    object-fit: cover;
+    object-position: center;
+    height: 100%;
+    max-width: 100%;
+}
+
+.block.floating-images .content {
+    border-left: 3px solid #e10;
```
Comment (bstopp) [inline blocks/floating-images/floating-images.css:32]: use variables for colors

File: blocks/floating-images/floating-images.css [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    margin-bottom: 1.5em;
+    padding: 0 0 16px;
+    line-height: var(--line-height-s);
+    font-size: var(--body-font-size-m);
+    font-weight: var(--font-weight-light);
+    color: var(--quartz);
+}
+
+/*** move-to-styles? ***/
+.block.floating-images .content h2 {
```
Comment (bstopp) [inline blocks/floating-images/floating-images.css:48]: if this is what *all* h2 will use, yes, we can move this to the global styles file.

File: blocks/floating-images/floating-images.css [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+.block.floating-images .content {
+    border-left: 3px solid #e10;
+    padding: 0 24px;
+    max-height: 350px;
+}
+
+/*** move-to-styles? ***/
+.block.floating-images .content p {
```
Comment (bstopp) [inline blocks/floating-images/floating-images.css:38]: If all paragraphs use this as the default experience, yes, we can move to global styles. However, the default font weight should never be light - should be 'normal'

**Review by bstopp — CHANGES_REQUESTED** [review]

File: blocks/floating-images/floating-images.css (hunk 5 above)
Comment (DanCollinz) [reply blocks/floating-images/floating-images.css:32]: fixed

File: blocks/floating-images/floating-images.css (hunk 6 above)
Comment (DanCollinz) [reply blocks/floating-images/floating-images.css:48]: moved to styles

File: blocks/floating-images/floating-images.css (hunk 7 above)
Comment (DanCollinz) [reply blocks/floating-images/floating-images.css:38]: moved

File: .gitignore (hunk 1 above)
Comment (DanCollinz) [reply .gitignore:10]: removed

File: scripts/lib-franklin.js [hunk 8]
```diff
@@ -129,6 +129,7 @@ export async function loadScript(src, attrs) {
  * @returns {string} The metadata value(s)
  */
 export function getMetadata(name) {
+  console.log(name);
```
[the remark this answers is not in the document]
Comment (DanCollinz) [reply scripts/lib-franklin.js]: fixed
